src/ucs/sys/netlink.h

```diff
@@ -45,20 +45,33 @@ ucs_netlink_send_request(int protocol, unsigned short nlmsg_type,
 
 
 /**
- * Check whether a routing table rule exists for a given network
- * interface name and a destination address.
+ * Check whether a specific route exists for a given network interface and
+ * destination address.
  *
- * @param [in]  if_index          A global index representing the network
-                                  interface, as assigned by the system
-                                  (e.g., obtained via if_nametoindex()).
- * @param [in]  sa_remote         Pointer to the destination address.
- * @param [in]  allow_default_gw  Allow matching default gateway routes (1) or
- *                                only specific subnet routes (0).
+ * @param [in]  if_index           A global index representing the network
+ *                                 interface, as assigned by the system
+ *                                 (e.g., obtained via if_nametoindex()).
+ * @param [in]  sa_remote          Pointer to the destination address.
+ * @param [out] has_default_gw_p   Optional pointer to store whether a default
```

Comment:
wdyt of: instead of returning 0/1 and an optional _p, have a single return value with an enum like `{None,Route,Gateway}` (roughly)?